services/bots.py

```diff
@@ -1,21 +1,39 @@
 import logging
-from random import shuffle
-from typing import Dict, Optional, Tuple
+import random
+from datetime import UTC, datetime
+from typing import Dict, List, Optional, Tuple
 
 from shared.config import get_config
 from shared.torngit.base import TokenType
 
 from database.models import Owner, Repository
-from database.models.core import GITHUB_APP_INSTALLATION_DEFAULT_NAME
+from database.models.core import (
+    GITHUB_APP_INSTALLATION_DEFAULT_NAME,
+    GithubAppInstallation,
+)
 from helpers.environment import is_enterprise
 from helpers.exceptions import OwnerWithoutValidBotError, RepositoryWithoutValidBotError
-from helpers.github_installation import get_installation_name_for_owner_for_task
 from services.encryption import encryptor
 from services.github import get_github_integration_token
 
 log = logging.getLogger(__name__)
 
 
+MAX_GITHUB_APP_SELECTION_WEIGHT = 1200
+
+
+def _get_installation_weight(installation: GithubAppInstallation) -> int:
+    """The weight for a given app installation.
+    Establishes an exponential ramp-up period for installations after being updated.
+    """
+    age = datetime.now(UTC) - installation.updated_at
+    if age.days >= 10:
+        return MAX_GITHUB_APP_SELECTION_WEIGHT
+    # seconds_in_hour = 3600
+    age_hours = (age.seconds // 3600) + age.days * 24
+    return age_hours + 2**age.days
```

Comment:
how did you decide what to do for this function?

Reply:
1st iteration it was going to be a linear function. The rampup was then gonna be about 50days, which is too long. And it makes more sense for it to be exponential, cause it starts slow and then ramps up fast.

So I discussed the requirements and acceptable parameters with Trent. We agreed on 1-2% in the first day and a rampo up period of 7-10 days. Then I plotted the graph in Desmos to figure out more or less how it was gonna look like.

Once I was satisfied I ran the test with choices many times to check the range of sleections for the new installation to see if it was within the parameters I wanted. Et voilà